Chessington.GameEngine/Pieces/Bishop.cs

```diff
@@ -10,7 +10,25 @@ public Bishop(Player player)
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
-            return Enumerable.Empty<Square>();
+            var currentPos = board.FindPiece(this);
+            var currentRow = currentPos.Row;
+            var currentCol = currentPos.Col;
+            List<Square> availableMoves = new List<Square>();
+            foreach (var col in Enumerable.Range(0, 8))
+            {
+                var columnsAway = col - currentCol;
+                if (Enumerable.Range(0, 8).Contains(currentRow + columnsAway))
```

Comment:
seems like more expensive way to check if the column is in bounds :) but it's fine for now